### PR TITLE
Start of the Data Module and Postgres Impl. Module -- Subject to change after the PostgreSQL Schema is finalized

### PR DESCRIPTION
**Pull request by zmoore-usgs**
Note: The mybatis xml still needs to have its column mapping fleshed out.

Also Note: This includes the initial webservice and MQ service layer pieces from PTs fork because I needed to use the DROP configuration loader he added. His PR should be merged before this one and then those changes won't show up as additions in my PR.

### CODE REVIEW

File: mlr-data-postgres-impl/src/main/resources/mybatis/postgresql/MonitoringLocationMapper.xml [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+		SELECT
+			<include refid="MonitoringLocationColumns"/>
+		FROM ML_ENTITY_TABLE
+			<include refid="whereAndOrder"/>
+	</select>
+	
+	<!--Where caluses for dealing with optional parameters and ordering-->
+	<sql id="whereAndOrder">
+		<where>
+			<if test="id != null">AND ID = '${id}'</if>
```
Comment (kmschoep-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/postgresql/MonitoringLocationMapper.xml]: Will have to explore how to do sql-injection-safe parameter substitution.

Comment (zmoore-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/postgresql/MonitoringLocationMapper.xml]: Yeah, this is how it is handled currently in the GDAWS-AQ-SYNC program we created for GCMRC but I'm not sure if that's the best way or not

File: mlr-data-postgres-impl/src/main/resources/mybatis/postgresql/MonitoringLocationMapper.xml [hunk 2]
```diff
@@ -0,0 +1,121 @@
...
+		SELECT
+			<include refid="MonitoringLocationColumns"/>
+		FROM monitoring_location
+			<include refid="whereAndOrder"/>
+	</select>
+	
+	<!--Where caluses for dealing with optional parameters and ordering-->
+	<sql id="whereAndOrder">
+		<where>
+			<if test="id != null">AND id = '${id}'</if>
```
Comment (kmschoep-usgs) [inline mlr-data-postgres-impl/src/main/resources/mybatis/postgresql/MonitoringLocationMapper.xml]: id is an integer field, so single quote not necessary, but more importantly, it should be `#{id}` to be a true parameter substitution, and thus sql-injection-safe.

**Comment by thongsav-usgs** [comment]
I was thinking that the mlr-data-api package would only be interfaces, so that other impls would know what has to be implemented.

So MonitoringLocationDAO.java would move to postgres-impl and implement an IMonitoringLocationDAO.java impl.

**Pull request merged**